test/unit/saas_file_mock.py

```diff
@@ -0,0 +1,89 @@
+from __future__ import annotations
+from typing import Iterable, ByteString, BinaryIO
+from datetime import datetime
+import os
+from io import IOBase
+import shutil
+import errno
+from pathlib import Path
+
+from exasol.bucketfs.saas_file_api import SaasFile, UNSET, SAAS_FOLDER
+
+
+class SaasFileApiMock:
```

Comment:
In terms of testing terminology, this is more accurately described as either a fake or a facade rather than a mock.